test/lib/agreement.js

```diff
@@ -25,44 +28,50 @@ contract('Agreement Interface', function(accounts) {
   };
 
   beforeEach(async () => {
-    market = await (new MarketRepository(MarketArtifacts)).create(accounts[0]);
-    offer = await (new OfferRepository(OfferArtifacts)).save(market.marketContract.address, testOffer);
-    token = await Token.new([web3.eth.accounts[0]], [1000]);
-    agreement = new Agreement(offer.address, 3, token.address, AgreementArtifacts, Token);
+    market = await (new MarketRepository()).create(1000);
+    offer = await (new OfferRepository()).save(market.marketContract.address, testOffer);
+    var agreementRepo = new AgreementRepository(market.getAddress(), AgreementArtifacts);
+    token = Token.at(await market.marketContract.token());
+    agreement = await agreementRepo.initiateAgreement(offer.address, 3);
+    agreementContract = agreement.agreementContract;
   });
 
   it('should setup buyer & seller', async () => {
-    var agreementContract = await agreement.initiateAgreement();
-
     assert.deepEqual(await agreementContract.buyer(), accounts[0]);
     assert.deepEqual(await agreementContract.seller(), accounts[1]);
   });
 
   it('should escrow', async () => {
-    var agreementContract = await agreement.initiateAgreement();
-
     assert.equal(await token.balanceOf(accounts[0]), 400);
     assert.equal(await token.balanceOf(agreementContract.address), 600);
   });
 
   it('should catch if not enough tokens', async () => {
```

Comment:
"should throw an exception if not enough tokens" not "catch an exception" cause we're checking if subject of test is throwing